StylableSwiftUI/Classes/Specificity.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  Specificity.swift
+//  StylableSwiftUI
+//
+//  Created by Sam Dean on 10/04/2020.
+//
+
+import Foundation
+
+extension StylistIdentifier {
+
+    /// A structure representing how specific a stylist identifier is.
+    ///
+    /// The actual value isn't useful (and is accordingly private), it's only useful when comparing with
+    /// other `Specificity` values.
+    struct Specificity {
+
+        private let value: UInt
+
+        init(components: [StylistIdentifier.Component]) {
+            let result = components.reversed().reduce((index: 0, score: UInt(0))) { result, component in
+                var result = result
+                result.index += 1
+                if component.value != nil {
+                    result.score |= 1 << result.index
```

Comment:
Why OR + bitshift instead of just `+= 1`?

Reply:
Because the importance of the component depends on its index. For example, if I just added 1 each time, these would be the same specificity (i.e. they would both have the score of 2):

```swift
"*/element/atom"
"section/*/atom"
```

However, we want the first one to be more specific than the second. So I use the index to make that importance. So, using the bit shift, the scores would become 6 and 5 respectively.

Identifier | Components | Total
---------- | ---------- | -----
`*/*/*` | 0 + 0 + 0 | 0
`*/*/atom` | 0 + 0 + 4 | 4
`*/element/*` | 0 + 2 + 0 | 2
`*/element/atom` | 0 + 2 + 4 | 6
`section/element/atom` | 1 + 2 + 4 | 7

This is making my brain hurt, so please let me know if I've got this completely wrong!

Oh, regarding `|=` vs. `+=` there's literally no difference in this case :) It's just good practice when using bit masks to use `OR` so you don't get overflow if you `+` and the bit is already set.

Reply:
@atomoil / @kerrmarin  Interesting question has fallen out of me making the table in the comment above - which of these two identifiers would you consider more specific (i.e. which one should the stylist match first?)

```
*/element[selected]/atom
section/element/atom
```

FYI: Currently, it's going to match `*/element[selected]/atom`.